### ISSUE TITLE
Visibility stuck on embargo in Save Work widget

### ISSUE THREAD

**Issue by mjgiarlo** [projecthydra-labs/hyku#971]
Details forthcoming. This is a stub so I remember to come back later and analyze what I'm seeing.

**Comment by mjgiarlo** [comment]
@tdonohue Can you take a look at this? I can reproduce it locally and on AWS. To reproduce... try uploading a work of any type and setting visibility to anything other than embargo. (I am depositing into the default admin set, which I have set up to allow release = varies.) FWIW, I'm seeing this line return tomorrow's date as `releaseDate`: https://github.com/projecthydra-labs/hyrax/blob/master/app/assets/javascripts/hyrax/save_work/visibility_component.es6#L59

**Comment by tdonohue** [comment]
@mjgiarlo : I cannot seem to reproduce this, but it's possible I'm misunderstanding the steps you've taken. Here's what I've tried (using latest Hyku `master` as of today).

1. Started with a fresh Hyku install and I'm using latest Chrome browser. Created the default AdminSet (via rake task)
2. Edited the Default AdminSet to set Release="Varies", and selected "Between now and 6 months after deposit" option.
3. Began depositing a new Work into that Default AdminSet. Added all metadata & files.
4. In the "Save Work" sidebar/widget, I see that all Release options are enabled *except for* "Lease".  I selected a non-embargo option (tried saving with both "Open Access" and "Private")
5. Checked the deposit agreement & Saved the work.
6. The work is saved successfully. If I go back and "Edit" the work, the "Save Work" sidebar/widget shows the Release option that I selected.

The behavior to me seems correct. I'm uncertain what you mean by "visibility stuck on embargo".  Are you unable to select another option (prior to saving the new work)?  Are you able to save the work, but it reverts to an embargo? Or, maybe I'm just overlooking something in my testing?

**Comment by mjgiarlo** [comment]
Have a look in my troubleshooting tenant. Here's how my AdminSet is configured:

![screenshot from 2017-04-20 19-45-59](https://cloud.githubusercontent.com/assets/131982/25260584/27ea2fbc-2602-11e7-983a-357039e69b4a.png)

And here's what I see when I go to deposit a work (in said AdminSet):

![screenshot from 2017-04-20 19-46-35](https://cloud.githubusercontent.com/assets/131982/25260601/3b9e024a-2602-11e7-9ca3-8ba7f1c9d7c8.png)

I can't select anything other than embargo. Maybe it's an artifact of our AWS-deployed installation, and will go away when we rebuild?

**Comment by tdonohue** [comment]
@mjgiarlo : Thanks for the screenshots, they were very helpful in figuring out what is going on.

After further analysis, I've found this is a bug, and I can reproduce it locally under a very specific scenario (so, excellent find!).

Essentially, it only seems to occur when ALL of the following is true:
* Selected AdminSet is configured with Release="No Delay" (i.e. release today)
* The date on the server is *different from* the date on the client (e.g. the server and the client are in different timezones and the date is actually different in those timezones).

So, I'm able to reproduce this locally by setting my server timezone to Brisbane time (where the date is currently Sat, April 22) and access the site from a client machine in a USA timezone (where the date is currently Fri, April 21).

Long story short, this is a bug in determining *today's date* between the code running on the server side and the code running on the client side. I'll dig in deeper now and work on a PR to fix this.

**Comment by mjgiarlo** [comment]
@tdonohue Wow, great find!

**Comment by tdonohue** [comment]
@mjgiarlo : Now has a PR: https://github.com/projecthydra-labs/hyrax/pull/820
